library/ssl_misc.h

```diff
@@ -987,6 +982,11 @@ struct mbedtls_ssl_handshake_params
     } tls13_master_secrets;
 
     mbedtls_ssl_tls13_handshake_secrets tls13_hs_secrets;
+#if defined(MBEDTLS_SSL_EARLY_DATA)
+    mbedtls_ssl_tls13_early_secrets tls13_early_secrets;
```

Comment:
As you mentioned there is no need for this to be in the handshake context. We can do the job with only local variables in ssl_tls13_keys.c. Thus I would be on the side of removing it but if you need it for a coming PR.

Reply:
#6675 is created for the topics.

Reply:
Is **tls13_early_secrets** needed in the handshake context in the future? if it isn't needed, prefer change it here.  Seems there isn't too much changes here.

Reply:
There are many relative changes. That will break current structure.  And relative unit tests must be change. 

I have tried that before pushing, that's not small loading task, I think we should  resolve that in #6675.

Reply:
ok let's do this in #6675